grishma/tests/acceptance/stepDefinitions/loginContext.js

```diff
@@ -0,0 +1,22 @@
+const {Given, When, Then} = require('@cucumber/cucumber')
+const { expect } = require("@playwright/test")
+var assert = require('assert')
+
+const LoginPage = require("../pageObjects/LoginPage.js")
+const loginPage = new LoginPage
+
+
+Given('the user has browsed to the login page', async function () {
+  await loginPage.gotoLoginPage()
+  await expect(page).toHaveURL(loginPage.url)
+});
+
+
+When('user logins with username {string} and password {string}', async function (username, password) {
+  await loginPage.logintoHomePage(username,password)
+});
+
+
+Then('user should redirect to the homepage', async function () {
+  await expect(page).toHaveURL(loginPage.homePageUrl)
+});
```

Comment:
this assertion seems a bit weird but from the user POV i would see something in the UI so that i can convinced that i am logged in. may be check some UI elements which is constant.